### ISSUE TITLE
Modelling transformers with asymmetric charged susceptance

### ISSUE THREAD

**Issue by eperim** [e2nIEE/pandapower#2461]
### Feature Checklist

- [X] Searched the [issues page](https://github.com/e2nIEE/pandapower/issues) for similar reports

- [X] Read the relevant sections of the [documentation](https://pandapower.readthedocs.io/en/latest/about.html)

- [X] Browse the [tutorials](https://github.com/e2nIEE/pandapower/tree/develop/tutorials) and [tests](https://github.com/e2nIEE/pandapower/tree/develop/pandapower/test) for usefull code snippets and examples of use

### Issue

Pandapower offers two approaches to modelling transformers: one is to use a symmetric pi-model, with equal susceptance values on each terminal; the other is to use a t-model, with the susceptance in the centre. Looking at the source code, it seems that both [models ultimately get translated into a symmetric pi-model](https://github.com/e2nIEE/pandapower/blob/869ec33d2a6afc7ee9ec9fca3d949b10c214808b/pandapower/build_branch.py#L353).

However, it is not uncommon for asymmetric models to be employed, particularly one in which the charged susceptance is applied only on the tap side. Is there any way to simulate this kind of model within Pandapower? Or does it fundamentally require using symmetric transformer models?

### Label

- [ ] Relevant labels are selected

**Comment by eperim** [comment]
@vogt31337 I am collaborating with people that use proprietary software which models transformers with the shunt only on the tap side, and that is introducing some significant discrepancies on the power flows I get using pandapower. Ideally, if I could specify two susceptance values, one on the HV side and another on the LV side, I'd be able to reproduce their results.

**Comment by pawellytaev** [comment]
Hey @eperim , currently pandapower doesn't support asymmetric admittances for trafos. However, we are planning to have this as a feature in the future, but haven't started it yet. In the current develop branch we have already added the `G_ASYM` and `B_ASYM` as additional columns to the `ppc['branch']` table which gives us the possibility to have this feature.

We have added asymmetric admittances to the impedance element https://pandapower.readthedocs.io/en/latest/elements/impedance.html#pandapower.create_impedance (new parameters `gf_pu`, `gt_pu`, `bf_pu`, `bt_pu`) where you have asymmetric admittances. So you can try and emulate your case with this one (only works for trafos with phase shift = 0, since impedance elements don't have a phase shift in pandapower)

**Comment by pawellytaev** [comment]
Do you see asymmetric admittances relevant both for 2w and 3w-trafos? I have personally only seen this in other software in the 3w-trafo model

**Comment by eperim** [comment]
@pawellytaev Thanks for the info. Unfortunately, I'm also dealing with transformers that are phase shifters. Think I'm out of luck =/

In the case I am dealing with, the admittance is asymmetric for both the 2w and the 3w transformers. I wouldn't know how common this is, though.

**Comment by pawellytaev** [comment]
Hm, do you know the use case of trafos with asymmetric admittances? I can't think of a specific reason why one would model them this way in a pi equivalent circuit.

**Comment by eperim** [comment]
I don't, but I am not a power systems (or electrical) engineer, so I am not really the best person to answer that. In the specific case I am dealing with, it's not just asymmetric, but it has no added shunt at all on the side opposite to the tap (not sure if that makes it more reasonable than modelling it with just a general asymmetric model or not, when it comes to the physics of the component).

**Comment by vogt31337** [comment]
@eperim, maybe you can provide a simple example with concrete values, so that we can later on test if our implementation is working with your use case?

**Comment by eperim** [comment]
@vogt31337 I could look into sharing one of the examples I used as reference, but those are in PSSE's RAW format. Would that work for you?

**Comment by vogt31337** [comment]
I personally do Not know that Format. I would prefer to have a short Code snippet, which created such an asymmetric Transformer, does a calculation, Like PP.runpp and the checks the results If they are correct.

**Comment by omalleyconor** [comment]
Hi, would the following snippet suffice? It calculates the flow in a transformer for different phase shifts but the magnetizing inductance shunt is only applied to one side of the transformer.
[Calculate_Flow.zip](https://github.com/user-attachments/files/19684718/Calculate_Flow.zip)

**Comment by pawellytaev** [comment]
Hey @omalleyconor, thanks for sharing the code snippet! Very helpful :-)

Just a few questions to check if my understanding is correct:
From what I understand from your code, the trafo model would look like this

![Image](https://github.com/user-attachments/assets/39d610f3-3e47-44e0-aeaf-40b8a9630b10)

(sorry for my bad MS Paint skills)

Is there also the possibility to model the asymmetric Y_sh somewhere else? e.g. at the LV side after the Z_br, or between the ideal transformer and the Z_br?


If yes, then this has the following implications on the pandapower internal structure:
**TL;DR**
It is generally possible to implement this feature in pandapower, but the exact way how to do it has pro's and con's. We will take this up internally in the dev team and come back to you here.

**Long version:**
Here are the possibilities I see to implement this in pandapower (for internal consideration):
Either 1)
We add the Y_sh as G&B to the `ppc['bus']` table - **Pro**: easy to implement, **Con**: but then we have trafos modelled both in `ppc['bus']` and `ppc['branch']`

Or 2)
We add an additional branch element in `ppc['branch']` with R&X=0 and G&B=Y_sh.real&Y_sh.imag and the trafo branch with R&X>0, G&B=0 - **Pro**: trafo only modeled in `ppc['branch']`, **Con** makes out of one trafo, 2 branch elements in `ppc['branch']`

We'll consider both options internally.


Cheers,
Pawel

**Comment by omalleyconor** [comment]
Hi @pawellytaev ,

Thanks for the thoughtful response, I think option 1 would make most sense. It would need to add some checks that the transformer "Flow" includes the shunt but this is relatively easy to do in post processing. 

Thanks
